File: SourceCode/XeKhachTools/src/main/java/data/entities/BussTrip.java
package data.entities;

import data.entities.abstracts.AbstractEntity;
import data.models.Column;
import data.models.MapColumn;
import lombok.Getter;
import lombok.Setter;

import static data.entities.abstracts.DataType.*;

@Getter @Setter
public class BussTrip extends AbstractEntity {
    {
        pk(Buss.class);
        pk(Company.class);
    }
    Column launchTime = of(TIME_ONLY);
    Column effectiveDateFrom = of(DATE_ONLY);
    Column price = of(MONEY);

    Column sunday = of(FALSE);
    Column monday = of(FALSE);
    Column tuesday = of(FALSE);
    Column wednesday = of(FALSE);
    Column thursday = of(FALSE);
    Column friday = of(FALSE);
    Column saturday = of(FALSE);

    MapColumn tripPoints = map(TripPoint.class);

    MapColumn startPoint = map(BussPoint.class);
    MapColumn endPoint = map(BussPoint.class);
}

File: SourceCode/XeKhachAngular/src/app/business/entities/buss-trip.ts
import {Buss} from "./buss";
import {BussPoint} from "./buss-point";

export class BussTrip {
  bussTripId: number;
  tripId: number;

  companyId: number;
  bussId: number;
  buss: Buss;


  startPointLocationId: number;
  startPointBussPointId: number;
  startPoint: BussPoint;

  endPointLocationId: number;
  endPointBusPointId: number;
  endPoint: BussPoint;

  launchTime: Date;
  effectiveDateFrom: Date;
  sunday: boolean;
  monday: boolean;
  tuesday: boolean;
  wednesday: boolean;
  thursday: boolean;
  friday: boolean;
  saturday: boolean;

  constructor() {
    this.endPoint = new BussPoint();
    this.startPoint = new BussPoint();
    this.buss = new Buss();
  }

}
